go/summarize/summarize.go

```diff
@@ -115,11 +117,66 @@ func printSummary(hotMetric string, workers []summaryWorker) (*Summary, error) {
 			return nil, err
 		}
 	}
-	err = s.PrintMarkdown(os.Stdout, time.Now())
+	outputFormat = strings.ToLower(outputFormat)
+	if *port == 0 && outputFormat == "html" {
+		fmt.Println("port is required when output format is html")
+		os.Exit(1)
+	}
+	switch outputFormat {
+	case "html":
+		summaryJSON, err := json.Marshal(*s)
+		if err != nil {
+			fmt.Println("Error marshalling summary:", err)
+			return nil, err
+		}
+
+		tmpFile, err := writeToTempFile(summaryJSON)
+		if err != nil {
+			return s, err
+		}
+		err = launchInBrowser(tmpFile)
+		if err != nil {
+			return s, err
+		}
+	case "markdown":
+		// Print the response
+		err = s.PrintMarkdown(os.Stdout, time.Now())
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, errors.New("unknown output format")
+	}
+	return s, nil
+}
+
+func launchInBrowser(tmpFile *os.File) error {
+	port := int64(8080) // FIXME: take this from flags
+	url := fmt.Sprintf("http://localhost:%d/summarize?file=", port) + tmpFile.Name()
+	err := exec.Command("open", url).Start()
+	if err != nil {
+		fmt.Println("Error launching browser:", err)
+		return err
+	}
+	fmt.Println("URL launched in default browser:", url)
+	return nil
+}
+
+func writeToTempFile(summaryJSON []byte) (*os.File, error) {
+	tmpFile, err := os.CreateTemp("/tmp/", "vt-summary-*.json")
 	if err != nil {
+		fmt.Println("Error creating temp file:", err)
 		return nil, err
 	}
-	return s, nil
+	defer tmpFile.Close()
+
+	_, err = tmpFile.WriteString(string(summaryJSON))
+	if err != nil {
+		fmt.Println("Error writing to temp file:", err)
+		return nil, err
+	}
+
+	return tmpFile, err
 }
```

Comment:
Yeah I think we could avoid this step by creating and starting the web server directly from here instead of creating temporary files that will be used by the server created beforehand. Moving the logic here will give us more control over what and how we render.